File: infinite-herbs-backend/app/models/permission.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db.database import Base
from .role_permissions import role_permissions

class Permission(Base):
    __tablename__ = "permissions"

    id = Column(Integer, primary_key=True, autoincrement=True)
    key = Column(String, unique=True, nullable=False, index=True)
    description = Column(String, nullable=True)
    createdAt = Column(DateTime(timezone=True), server_default=func.now())
    updatedAt = Column(DateTime(timezone=True), onupdate=func.now())

    roles = relationship(
        "Role",
        secondary=role_permissions,
        back_populates="permissions",
    )

File: infinite-herbs-backend-.net/Models/Permission.cs
using System.ComponentModel.DataAnnotations;
namespace AdminApi.Models;
public class Permission {
    public int Id { get; set; }
    [Required] public string Key { get; set; } = "";
    public string Description { get; set; } = "";
    public ICollection<RolePermission> Roles { get; set; } = new List<RolePermission>();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
